djangocms_versioning/managers.py

```diff
@@ -78,12 +78,68 @@ def current_content(self, **kwargs):
             .values_list("vers_pk", flat=True)
         return qs.filter(versions__pk__in=pk_filter)
 
+    def latest_content(self, **kwargs):
+        """Returns the "latest" content object which is in this order
+           1. a draft version (should it exist)
+           2. a published version (should it exist)
+           3. any other version with the highest pk
+
+        This filter assumes that there can only be one draft created and that the draft as
+        the highest pk of all versions (should it exist).
+        """
+        """
+        While this is probably to most general approach it does not work for MySql :-(
+        inner = (
+                    self.annotate(
+                        order=models.Case(
+                            models.When(versions__state=constants.PUBLISHED, then=2),
+                            models.When(versions__state=constants.DRAFT, then=1),
+                            default = 0,
+                            output_field = models.IntegerField(),
+                        ),
+                        modified =  models.Max("versions__modified"),
+                    )
+                    .filter(**{
+                        self._group_by_key[0]: models.OuterRef(self._group_by_key[0])
+                    })
+                    .order_by("-order", "-modified")
+        )
+        return self.filter(pk__in=models.Subquery(inner[:1].values("pk")))
+        """
+        current = self.filter(versions__state__in=(constants.DRAFT, constants.PUBLISHED))\
+            .values(*self._group_by_key)\
+            .annotate(vers_pk=models.Max("versions__pk"))
+        pk_current = current.values("vers_pk")
+        pk_other = self.exclude(**{key + "__in": current.values(key) for key in self._group_by_key})\
+            .values(*self._group_by_key)\
+            .annotate(vers_pk=models.Max("versions__pk"))\
+            .values("vers_pk")
+        return self.filter(versions__pk__in=pk_current | pk_other, **kwargs)
+
 
 class AdminManagerMixin:
```

Comment:
I can;t see test coverage for this Mixin, was it added in a previous PR that I haven't seen?

Reply:
See here: https://github.com/django-cms/djangocms-versioning/blob/edfb62c8202105f3a37c27aa7e2f244567f0d7f9/tests/test_content_models.py#L22-L30 and following tests using `admin_manager`